winch/codegen/src/codegen/mod.rs

```diff
@@ -1005,9 +1005,9 @@ where
         base: Reg,
         table_data: &TableData,
     ) -> Result<M::Address> {
-        let scratch = scratch!(M);
         let bound = self.context.any_gpr(self.masm)?;
         let tmp = self.context.any_gpr(self.masm)?;
+        let tmp2 = self.context.any_gpr(self.masm)?;
```

Comment:
One thing to note here is that this will potentially increase register pressure for x64, but in reality we're trying to fix an Aarch64 issue with immediate value handling. I don't think this is a blocker, but wanted to point this out since this will likely change in the near-future since I have a work-in-progress branch which fixes the scratch register handling more broadly and will hopefully allow us to keep the previous implementation for x64.

Reply:
Oh nice! Would you prefer to hold off on landing this until said refactoring has landed?

Reply:
Sounds good to me, yeah.

Reply:
I have part 2/N here https://github.com/bytecodealliance/wasmtime/pull/10989, once that one lands, I think the only remaining piece is to ensure that any scratch register usage in ISA-agnostic goes through the proper path to allocate scratch registers and then we could land this one. Said differently, I think we're 1 change away from unblocking this.